### PR TITLE
[WIP] Epic branch for accessibility

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR is an epic branch that contains different things that improve the accessibility of the component. The branch currently contains the following changes:

ARIA tags:
 * https://github.com/handsontable/handsontable/pull/10463
 * https://github.com/handsontable/handsontable/pull/10524
 * https://github.com/handsontable/handsontable/pull/10526
 * https://github.com/handsontable/handsontable/pull/10535
 * https://github.com/handsontable/handsontable/pull/10539
 * https://github.com/handsontable/handsontable/pull/10530
 * https://github.com/handsontable/handsontable/pull/10548
 * https://github.com/handsontable/handsontable/pull/10541

Support for new keyboard shortcuts:
 * https://github.com/handsontable/handsontable/pull/10237
 * https://github.com/handsontable/handsontable/pull/10389
 * https://github.com/handsontable/handsontable/pull/10404
 * https://github.com/handsontable/handsontable/pull/10405
 * https://github.com/handsontable/handsontable/pull/10410
 * https://github.com/handsontable/handsontable/pull/10417
 * https://github.com/handsontable/handsontable/pull/10413
 * https://github.com/handsontable/handsontable/pull/10420
 * https://github.com/handsontable/handsontable/pull/10419
 * https://github.com/handsontable/handsontable/pull/10421
 * New behavior and keyboard shortcuts for the context and dropdown menu (https://github.com/handsontable/handsontable/pull/10519)

New features:
 * https://github.com/handsontable/handsontable/pull/10301
 * https://github.com/handsontable/handsontable/pull/10430
 * https://github.com/handsontable/handsontable/pull/10342
 * https://github.com/handsontable/handsontable/pull/10529

Other changes:
 * https://github.com/handsontable/handsontable/pull/10265
 * https://github.com/handsontable/handsontable/pull/10464
 * New hooks and breaking changes in the `scrollViewportTo` method (https://github.com/handsontable/handsontable/pull/10508)
 * New `beforeSelectionHighlightSet ` hook (https://github.com/handsontable/handsontable/pull/10513);
 * https://github.com/handsontable/handsontable/pull/10520

_[skip changelog]_

### Affected project(s):
- [x] `handsontable`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

**Comment by aninde** [comment]
@budnix there is regression on this branch, found by @krzysztofspilka. 

Increasing the selection range by shortcuts does not work for adding different axes, from vertical to horizontal and vice verse. 
Steps to reproduce: 
On demo https://stackblitz.com/edit/js-4w79ps?file=style.css,index.js  build this PR select cell D3 and 
1. press `Cmd + Shift + Arrow down` and hold `Cmd + Shift`
2. Press `Arrow left`

https://github.com/handsontable/handsontable/assets/30186818/2169c78d-00d7-410e-b006-5fa268a89312

V13.0.0 without this issue  for comparision https://stackblitz.com/edit/js-gr18hr?file=style.css,index.js

https://github.com/handsontable/handsontable/assets/30186818/1bb066da-040a-43a9-be84-a33a040c67d1

File: .changelogs/10237.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added Ctrl/Cmd+Shift+Space keyboard shortcut that allows selecting all cells.",
```
Comment (kirszenbaum) [inline .changelogs/10237.json:3]:
```suggestion
  "title": "Added a keyboard shortcut for selecting all cells.",
```

File: .changelogs/10265.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added support for keyboard navigation in headers.",
```
Comment (kirszenbaum) [inline .changelogs/10265.json:3]:
```suggestion
  "title": "Added support for keyboard navigation in row and column headers.",
```

File: .changelogs/10389.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added support for keyboard shortcuts that allows selecting all rows and columns and add 4 new hooks",
```
Comment (kirszenbaum) [inline .changelogs/10389.json:3]:
```suggestion
  "title": "Added a keyboard shortcut for selecting all rows and columns. Added 4 new hooks.",
```

File: .changelogs/10404.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added support for collapsing/expanding columns using a keyboard shortcut.",
```
Comment (kirszenbaum) [inline .changelogs/10404.json:3]:
```suggestion
  "title": "Added a keyboard shortcut for collapsing or expanding columns.",
```

File: .changelogs/10405.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added support for collapsing/expanding nested rows using a keyboard shortcut.",
```
Comment (kirszenbaum) [inline .changelogs/10405.json:3]:
```suggestion
  "title": "Added a keyboard shortcut for collapsing or expanding nested rows.",
```

File: .changelogs/10410.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added support for column sorting using a keyboard shortcut.",
```
Comment (kirszenbaum) [inline .changelogs/10410.json:3]:
```suggestion
  "title": "Added a keyboard shortcut for rows sorting.",
```

File: .changelogs/10413.json [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added support for keyboard shortcuts that allow open column menu.",
```
Comment (kirszenbaum) [inline .changelogs/10413.json:3]:
```suggestion
  "title": "Added a keyboard shortcut for opening the column menu.",
```

File: .changelogs/10417.json [hunk 8]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added support for keyboard shortcut that allows opening the context menu.",
```
Comment (kirszenbaum) [inline .changelogs/10417.json:3]:
```suggestion
  "title": "Added a keyboard shortcut for opening the context menu.",
```

File: .changelogs/10419.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added support for new keyboard shortcut that allows scrolling the viewport",
```
Comment (kirszenbaum) [inline .changelogs/10419.json:3]:
```suggestion
  "title": "Added a new keyboard shortcut for scrolling the viewport.",
```

File: .changelogs/10420.json [hunk 10]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added support for keyboard shortcut that allows clearing the filters.",
```
Comment (kirszenbaum) [inline .changelogs/10420.json:3]:
```suggestion
  "title": "Added a keyboard shortcut for clearing all filters.",
```

File: .changelogs/10421.json [hunk 11]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added a keyboard shortcut that allows managing the cell comments.",
```
Comment (kirszenbaum) [inline .changelogs/10421.json:3]:
```suggestion
  "title": "Added keyboard shortcuts for managing cell comments.",
```

File: .changelogs/10430.json [hunk 12]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "private",
+  "title": "Added the ability to traverse the grid using TAB key throughout the page and added new `disableTabNavigation` option.",
```
Comment (kirszenbaum) [inline .changelogs/10430.json:3]:
```suggestion
  "title": "Added the ability to traverse the grid using the <kbd>Tab</kbd> key throughout the page. Also, added a new configuration option: `disableTabNavigation`.",
```

**Review by kirszenbaum — CHANGES_REQUESTED** [review]

**Pull request merged**